### PR TITLE
fix: filterable default value

### LINKED ISSUE #4612: [Bug] By default filter is visible and filterable property is disabled after drag and drop of select/drop-down widget
### Steps to reproduce the behaviour:

1. Drag and drop select/drop-down widget on canvas
2. Click on the widget to see the default drop-down options
3. Observe that the filter is visible
4. Now observe that in property pane the filterable property is disabled

Environment: Release

Screenshot 2021-05-21 at 11 01 39 AM

### PR DESCRIPTION
**Pull request by Tooluloope**
Fixes #4612  

## Type of change

- Update (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/dropdown-filterable-default-value 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.16 **(0.01)** | 36.19 **(0.01)** | 32.84 **(0.03)** | 54.65 **(0.01)**
 :green_circle: | app/client/src/utils/WidgetPropsUtils.tsx | 65.3 **(0.55)** | 58.64 **(0.03)** | 55.95 **(1.63)** | 64.58 **(0.52)**</details>

### CODE REVIEW

**Comment by somangshu** [comment]
I still don't see it fixed on the Deploy preview 

Screenshot 2021-05-25 at 4 35 44 PM

This probably requires a migration.

**Comment by Tooluloope** [comment]
> This probably requires a migration.

@somangshu added migrations.
Do confirm if it's optimal

**Review by somangshu — DISMISSED** [review]
Approving to run tests, Works on deploy preview

File: app/client/src/utils/WidgetPropsUtils.tsx [hunk 1]
```diff
@@ -738,9 +738,36 @@ const transformDSL = (currentDSL: ContainerWidgetProps<WidgetProps>) => {
     currentDSL.version = LATEST_PAGE_VERSION;
   }
 
+  if (currentDSL.version === 20) {
+    currentDSL = migrateFilterValueForDropDownWidget(currentDSL);
+    currentDSL.version = 21;
```
Comment (riodeuno) [inline app/client/src/utils/WidgetPropsUtils.tsx:743]: Let's update the LATEST_PAGE_VERSION here by updating the value of LATEST_PAGE_VERSION.
@marks0351 Please advice on how this is to be done.

**Review by riodeuno — CHANGES_REQUESTED** [review]
One comment, the rest looks good.

File: app/client/src/utils/WidgetPropsUtils.tsx (hunk 1 above)
Comment (marks0351) [reply app/client/src/utils/WidgetPropsUtils.tsx:743]: @Tooluloope can u pls increment LATEST_PAGE_VERSION to 21. the idea is to have a constant that holds the latest page version.

**Review by marks0351 — CHANGES_REQUESTED** [review]

**Review by somangshu — APPROVED** [review]

**Review by riodeuno — APPROVED** [review]
Approving for tests to run.

**Comment by somangshu** [comment]
@riodeuno is the code looks fine, Kindly let me know. We need to push this into the release as the feature has already gone to prod a couple weeks ago

**Review by somangshu — APPROVED** [review]
Approving to run tests

File: app/client/src/utils/WidgetPropsUtils.tsx [hunk 2]
```diff
@@ -761,17 +761,44 @@ const transformDSL = (currentDSL: ContainerWidgetProps<WidgetProps>) => {
...
   }
 
-  if (currentDSL.version === 22) {
+  if (currentDSL.version === 23) {
     currentDSL = addLogBlackListToAllListWidgetChildren(currentDSL);
-    currentDSL.version = 23;
+    currentDSL.version = 24;
+  }
+
+  if (currentDSL.version === 24) {
```
Comment (marks0351) [inline app/client/src/utils/WidgetPropsUtils.tsx:772]: Thanks for cleaning this up @Tooluloope .

**Review by marks0351 — APPROVED** [review]

**Comment by somangshu** [comment]
@Tooluloope is the latest release merged, Some failing test fixes are available.

**Review by somangshu — APPROVED** [review]

**Comment by somangshu** [comment]
@Tooluloope latest merge required here, Tests fixes are available

**Review by somangshu — APPROVED** [review]

**Comment by somangshu** [comment]
@Tooluloope please fix these merge conflicts and merge latest release here!

**Review by somangshu — APPROVED** [review]

**Comment by somangshu** [comment]
@Tooluloope latest release merge required for failing tests

**Review by somangshu — DISMISSED** [review]

**Review by somangshu — APPROVED** [review]

**Pull request merged**